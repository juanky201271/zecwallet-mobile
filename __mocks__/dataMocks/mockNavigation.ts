import { StackScreenProps } from '@react-navigation/stack';

export const mockNavigation: StackScreenProps<any>['navigation'] = {
  // Propiedades y métodos necesarios para la navegación
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
  reset: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  // Agrega cualquier otra propiedad o método necesario para tu caso
};
