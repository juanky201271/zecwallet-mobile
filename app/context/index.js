import { defaultAppContextLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppContextLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppContextLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppContextLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
