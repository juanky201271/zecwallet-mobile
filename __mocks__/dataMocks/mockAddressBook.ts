import { AddressBookFileClass } from '../../app/AppState';

export const mockAddressBook: AddressBookFileClass[] = [
  {
    label: 'pepe',
    address: 'u1234567890_____________',
  },
  {
    label: 'lolo',
    address: 'u0987654321_____________',
  },
];
