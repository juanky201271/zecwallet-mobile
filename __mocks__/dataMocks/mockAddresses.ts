import { AddressClass, AddressKindEnum, ReceiverEnum } from '../../app/AppState';

export const mockAddresses: AddressClass[] = [
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'UA-12345678901234567890',
    addressKind: AddressKindEnum.u,
    containsPending: false,
    receivers: ReceiverEnum.o + ReceiverEnum.z + ReceiverEnum.t,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'sapling-12345678901234567890',
    addressKind: AddressKindEnum.z,
    containsPending: false,
    receivers: ReceiverEnum.z,
  },
  {
    uaAddress: 'UA-12345678901234567890',
    address: 'transparent-12345678901234567890',
    addressKind: AddressKindEnum.t,
    containsPending: false,
    receivers: ReceiverEnum.t,
  },
];
