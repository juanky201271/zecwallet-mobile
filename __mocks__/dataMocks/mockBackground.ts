import { BackgroundType } from '../../app/AppState';

export const mockBackground: BackgroundType = {
  batches: 0,
  message: '',
  date: 0,
  dateEnd: 0,
};
