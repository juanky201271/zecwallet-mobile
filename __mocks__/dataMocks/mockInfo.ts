import { ChainNameEnum, CurrencyNameEnum, InfoType } from '../../app/AppState';
import { serverUris } from '../../app/uris';

export const mockInfo: InfoType = {
  chainName: ChainNameEnum.mainChainName,
  serverUri: serverUris(() => {})[0].uri,
  latestBlock: 2000000,
  connections: 0,
  version: 'server_version',
  verificationProgress: 0,
  currencyName: CurrencyNameEnum.ZEC,
  solps: 0,
  zingolib: 'zingolib_version',
};
