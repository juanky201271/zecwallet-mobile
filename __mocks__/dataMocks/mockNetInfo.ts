import { NetInfoStateType } from '@react-native-community/netinfo';
import { NetInfoType } from '../../app/AppState';

export const mockNetInfo: NetInfoType = {
  isConnected: true,
  type: NetInfoStateType.wifi,
  isConnectionExpensive: false,
};
