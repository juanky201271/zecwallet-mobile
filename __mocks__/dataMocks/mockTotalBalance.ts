import { TotalBalanceClass } from '../../app/AppState';

export const mockTotalBalance: TotalBalanceClass = {
  transparentBal: 0.12345678,
  privateBal: 0.4,
  spendablePrivate: 0.2,
  orchardBal: 0.6,
  spendableOrchard: 0.3,
  total: 1.12345678,
};
