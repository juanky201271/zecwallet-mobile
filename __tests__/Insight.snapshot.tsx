/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Insight from '../components/Insight';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockInfo } from '../__mocks__/dataMocks/mockInfo';
import { mockTotalBalance } from '../__mocks__/dataMocks/mockTotalBalance';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-svg-charts');
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component Insight - test', () => {
  //snapshot test
  test('Insight - snapshot', () => {
    const state = defaultAppContextLoaded;
    state.translate = mockTranslate;
    state.info = mockInfo;
    state.totalBalance = mockTotalBalance;
    const onClose = jest.fn();
    const insight = render(
      <ContextAppLoadedProvider value={state}>
        <Insight closeModal={onClose} openModal={onClose} setPrivacyOption={onClose} setSendPageState={onClose} />
      </ContextAppLoadedProvider>,
    );
    expect(insight.toJSON()).toMatchSnapshot();
  });
});
