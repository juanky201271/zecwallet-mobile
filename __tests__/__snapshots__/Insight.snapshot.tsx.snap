// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Insight - test Insight - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "display": "flex",
          "minHeight": 25,
          "paddingBottom": 0,
          "paddingTop": 10,
          "zIndex": -1,
        }
      }
      testID="header"
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginTop": 12,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginHorizontal": 5,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgb(255, 255, 255)",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 0,
                  "marginRight": 5,
                  "minHeight": 25,
                  "minWidth": 25,
                  "padding": 0,
                }
              }
            >
              <
                icon={
                  {
                    "icon": [
                      576,
                      512,
                      [],
                      "f3c1",
                      "M352 144c0-44.2 35.8-80 80-80s80 35.8 80 80l0 48c0 17.7 14.3 32 32 32s32-14.3 32-32l0-48C576 64.5 511.5 0 432 0S288 64.5 288 144l0 48L64 192c-35.3 0-64 28.7-64 64L0 448c0 35.3 28.7 64 64 64l320 0c35.3 0 64-28.7 64-64l0-192c0-35.3-28.7-64-64-64l-32 0 0-48z",
                    ],
                    "iconName": "lock-open",
                    "prefix": "fas",
                  }
                }
                size={25}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "left": 0,
          "padding": 11.5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
          }
        }
      />
    </View>
    <View
      style={
        {
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 38,
            "resizeMode": "contain",
            "width": 38,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="rgb(0, 122, 255)"
          icon={
            {
              "icon": [
                320,
                512,
                [
                  9001,
                ],
                "f053",
                "M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          size={30}
          style={
            {
              "marginHorizontal": 10,
            }
          }
        />
      </View>
      <Text
        selectable={true}
        style={
          {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        text translated
      </Text>
      <View
        style={
          {
            "height": 30,
            "marginHorizontal": 10,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomColor": "rgb(0, 122, 255)",
            "borderBottomWidth": 2,
            "paddingBottom": 10,
            "width": 243.33333333333334,
          }
        }
      >
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
              "fontWeight": "bold",
              "opacity": 1,
            }
          }
        >
          text translated
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(0, 122, 255)",
              "fontSize": 11,
              "fontWeight": "600",
              "opacity": 1,
            }
          }
        >
          (
          text translated
          )
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomColor": "rgb(0, 122, 255)",
            "borderBottomWidth": 0,
            "paddingBottom": 10,
            "width": 243.33333333333334,
          }
        }
      >
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 14,
              "fontWeight": "normal",
              "opacity": 1,
            }
          }
        >
          text translated
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 11,
              "fontWeight": "600",
              "opacity": 1,
            }
          }
        >
          (
          text translated
          )
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomColor": "rgb(0, 122, 255)",
            "borderBottomWidth": 0,
            "paddingBottom": 10,
            "width": 243.33333333333334,
          }
        }
      >
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 14,
              "fontWeight": "normal",
              "opacity": 1,
            }
          }
        >
          text translated
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 11,
              "fontWeight": "600",
              "opacity": 1,
            }
          }
        >
          (
          text translated
          )
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={{}}
    indicatorStyle="white"
    persistentScrollbar={true}
    showsVerticalScrollIndicator={true}
    style={
      {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "display": "flex",
            "margin": 20,
          }
        }
      >
        <ActivityIndicator
          color="rgb(0, 122, 255)"
          size="large"
          style={
            {
              "marginTop": 100,
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "marginHorizontal": 5,
            "padding": 0,
          }
        }
      >
        <View
          style={
            {
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
